### PR TITLE
new surface function

### PR DESCRIPTION
**Pull request by PDN96**
Modification made in the surface function. It used to crash while adding colors for an unordered array of vertices. Fixed the bug.

### CODE REVIEW

File: fury/actor.py [hunk 1]
```diff
@@ -188,6 +189,85 @@ def copy(self):
...
+        tri = Delaunay(xy)
+        faces = tri.simplices
+
+    count = 0
+    for face in faces:
+        p_1 = vertices[face[0]]
+        p_2 = vertices[face[1]]
+        p_3 = vertices[face[2]]
+
+        points.InsertNextPoint(p_1[0], p_1[1], p_1[2])
```
Comment (Garyfallidis) [inline fury/actor.py]: These loops can be avoided. Look here https://github.com/fury-gl/fury/blob/master/fury/utils.py#L351

**Comment by Garyfallidis** [comment]
Needs docstrings, test and tutorial. @PDN96 happy to help. Or you can ask @skoudoro or @dmreagan for suggestions in building the tests and examples.

**Comment by PDN96** [comment]
OK, I'll ask them and do the same! Also I'll look into how to make it more efficient using those functions.

**Comment by dmreagan** [comment]
Looks cool. It would be good to add an example to _viz_surfaces.py_.

**Comment by PDN96** [comment]
@Garyfallidis removed the all the loops and added documentation for the paramters.

File: fury/tests/test_actors.py [hunk 2]
```diff
@@ -153,6 +153,36 @@ def test_slicer():
...
+    for coordinate in vertices:
+        xy.append([coordinate[0], coordinate[1]])
+    tri = Delaunay(xy)
+    faces = tri.simplices
+    renderer = window.renderer(background=(1, 1, 1))
+    surface_actor = surface(vertices, smooth="butterfly")
+    axes_actor = actor.axes(scale=(12, 12, 12))
+    renderer.add(axes_actor)
+    renderer.add(surface_actor)
+    window.show(renderer, size=(600, 600), reset_camera=False)
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: The function `show` will block all the tests. you have to use `render` to save it as png file. Then load this png and check if we obtain the result as expected. You can look at the other test and let us know if you need help

**Comment by PDN96** [comment]
@skoudoro I was able to fix the pep8 and Codacy/PR Quality reviews but I can't seem to understand the Travis CI

File: fury/tests/test_actors.py [hunk 3]
```diff
@@ -154,6 +154,39 @@ def test_slicer():
...
+
+    random.shuffle(vertices)
+    vertices = np.array(vertices)
+    # xy = list()
+    # for coordinate in vertices:
+    #     xy.append([coordinate[0], coordinate[1]])
+    # tri = Delaunay(xy)
+    # faces = tri.simplices
+    renderer = window.renderer(background=(1, 1, 1))
+    surface_actor = surface(vertices, smooth="butterfly")
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: you error come from here @PDN96. this is `actor.surface`  and not `surface` only.
if you want to use `surface` only, you have to import it: `from fury.actor import surface`

**Comment by PDN96** [comment]
@skoudoro There's already a file called viz_surfaces.py in the fury/doc/examples folder. In my opinion, that should be renamed or if we don't want to do that please could you suggest a name for my example?
Thanks!

**Comment by PDN96** [comment]
@skoudoro I have updated the PR. Does that look okay?

File: fury/tests/test_actors.py [hunk 4]
```diff
@@ -154,6 +153,127 @@ def test_slicer():
...
+            fact2 = - math.exp(abs(1 - math.sqrt(i ** 2 + j ** 2) / math.pi))
+            z_coord = -abs(fact1 * fact2)
+            vertices.append([i, j, z_coord])
+
+    c_arr = np.random.rand(len(vertices), 3)
+    random.shuffle(vertices)
+    vertices = np.array(vertices)
+    tri = Delaunay(vertices[:, [0, 1]])
+    faces = tri.simplices
+    renderer_1 = window.renderer(background=(1, 1, 1))
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: Can you use `window.Scene` instead ?

File: fury/actor.py [hunk 5]
```diff
@@ -188,6 +188,95 @@ def copy(self):
     return image_actor
 
 
+def surface(vertices, faces=None, colors=None, smooth=None, subdivision=3):
+
```
Comment (skoudoro) [inline fury/actor.py]: Can you remove the empty space here

File: fury/tests/test_actors.py [hunk 6]
```diff
@@ -154,6 +158,43 @@ def test_slicer():
...
+    faces = tri.simplices
+
+    c_loop = [None, c_arr]
+    f_loop = [None, faces]
+    s_loop = [None, "butterfly", "loop"]
+
+    for smooth_type in s_loop:
+        for face in f_loop:
+            for color in c_loop:
+                renderer = window.renderer(background=(1, 1, 1))
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: Can you use `Scene` instead of `renderer`.

File: fury/tests/test_actors.py [hunk 7]
```diff
@@ -154,6 +158,43 @@ def test_slicer():
...
 
 
+@npt.dec.skipif(skip_it)
+@xvfb_it
+def test_surface():
+    size = 11
+    vertices = list()
+    for i in range(-size, size):
+        for j in range(-size, size):
+            fact1 = - math.sin(i) * math.cos(j)
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: you forgot to import `math` on the top

File: fury/tests/test_actors.py [hunk 8]
```diff
@@ -154,6 +158,43 @@ def test_slicer():
...
+    faces = tri.simplices
+
+    c_loop = [None, c_arr]
+    f_loop = [None, faces]
+    s_loop = [None, "butterfly", "loop"]
+
+    for smooth_type in s_loop:
+        for face in f_loop:
+            for color in c_loop:
+                renderer = window.Scene(background=(1, 1, 1))
```
Comment (skoudoro) [inline fury/tests/test_actors.py]: Can you replace `renderer` by `scene` everywhere?

Thank you!

**Comment by skoudoro** [comment]
Thank you @PDN96 for this work!

**Pull request merged**